test/sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncClientResetIntegrationTests.kt

```diff
@@ -0,0 +1,642 @@
+                    resume()
+
+                    // Validate we receive logs on the regular path
+                    assertEquals(ClientResetLogEvents.DISCARD_LOCAL_ON_BEFORE_RESET, logChannel.receive())
+                    assertEquals(ClientResetLogEvents.DISCARD_LOCAL_ON_AFTER_RESET, logChannel.receive())
+
+                    simulateError(
+                        ProtocolClientErrorCode.RLM_SYNC_ERR_CLIENT_AUTO_CLIENT_RESET_FAILURE,
+                        SyncErrorCodeCategory.RLM_SYNC_ERROR_CATEGORY_CLIENT
+                    )
+                    // Validate that we receive logs on the error callback
+                    assertEquals(ClientResetLogEvents.DISCARD_LOCAL_ON_ERROR, logChannel.receive())
+                }
+            }
+        }
+    }
+
+    @Test
+    fun defaultDiscardUnsyncedLocalChanges_flexible_logsReported() {
+        val config = SyncConfiguration.Builder(
+            user,
+            schema = setOf(FlexParentObject::class) // Use a class that is present in the server's schema
+        ).build()
+
+        Realm.open(config).use { realm ->
+            runBlocking {
+                // This channel helps to validate that the Realm gets updated
+
+                with(realm.syncSession as SyncSessionImpl) {
+                    simulateError(
+                        ProtocolClientErrorCode.RLM_SYNC_ERR_CLIENT_AUTO_CLIENT_RESET_FAILURE,
+                        SyncErrorCodeCategory.RLM_SYNC_ERROR_CATEGORY_CLIENT
+                    )
+                    // Validate that we receive logs on the error callback
+                    assertEquals(ClientResetLogEvents.MANUAL_ON_ERROR, logChannel.receive())
+                }
+            }
+        }
+    }
+
+    // Check that we can execute the Client Reset in a discard local strategy.
+    @Test
+    fun errorHandler_discardLocalExecuteClientReset() = runBlocking {
+        val channel = Channel<ClientResetEvents>(1)
+
+        val config = SyncConfiguration.Builder(
+            user,
+            partitionValue,
+            schema = setOf(FlexParentObject::class) // Use a class that is present in the server's schema
+        ).syncClientResetStrategy(object : DiscardUnsyncedChangesStrategy {
+            override fun onBeforeReset(realm: TypedRealm) {
+                fail("Should not call onBeforeReset")
+            }
+
+            override fun onAfterReset(before: TypedRealm, after: MutableRealm) {
+                fail("Should not call onAfterReset")
+            }
+
+            override fun onError(session: SyncSession, exception: ClientResetRequiredException) {
+                val originalFilePath = assertNotNull(exception.originalFilePath)
+                val recoveryFilePath = assertNotNull(exception.recoveryFilePath)
+                assertTrue(fileExists(originalFilePath))
+                assertFalse(fileExists(recoveryFilePath))
+
+                exception.executeClientReset()
```

Comment:
Yes, Core doesn't check for open instances. We can't unfortunately do what we do in Java because we lack a cache :( I will document the function making sure we highlight users must close all Realm instances to avoid file system-related crashes.